src/event-listeners/index.js

```diff
@@ -108,14 +112,60 @@ export default function setupEventListeners(store, canvas) {
   // kill performance
   canvas.addEventListener('mousemove', event => {
     const state = store.getState()
+    const { goals, edges, ui: { viewport: { translate, scale }, screensize: { width } }} = state
     if (state.ui.mouse.mousedown) {
-      store.dispatch(changeTranslate(event.movementX, event.movementY))
-      return
+     
+      if(event.shiftKey){
+        if(boolean===false){
+          x=event.clientX
+          y=event.clientY
+           convertedClick = coordsPageToCanvas({
+            x: event.clientX,
+            y: event.clientY
+        }, translate, scale)
+          boolean=true
+        }
+        
+         let lienzo = canvas.getContext("2d")
+         store.dispatch(unhoverGoal())
+         const convertedIni = coordsPageToCanvas({
+          x: event.clientX,
+          y: event.clientY
+      }, translate, scale)
+         let xd=convertedIni.x-convertedClick.x
+         let yd=convertedIni.y-convertedClick.y
+
+         lienzo.beginPath()
+        
+        lienzo.moveTo(convertedClick.x,convertedClick.y)
+        lienzo.lineTo(convertedClick.x+xd,convertedClick.y)
+        lienzo.lineTo(convertedClick.x+xd,convertedClick.y+yd)
+        lienzo.lineTo(convertedClick.x,convertedClick.y+yd)
+        lienzo.lineTo(convertedClick.x,convertedClick.y)
+
+        lienzo.closePath()
+        lienzo.stroke()
```

Comment:
all drawing on the canvas should happen in the `src/drawing/index.js` file, or at least originate in the `render` function of that file, if the code is written in a separate file, such as `drawGoal.js`. I would create a new file under `src/drawing` called `drawSelectBox`, where this code could go, and be exported from.

Reply:
Take a look at `drawGoalCard.js` for reference

Reply:
Actually @hector0513 , take a close look at `roundRect` function in `drawGoalCard.js`. It is probably exactly what you want! Just export it from that file and import it into `drawSelectBox`

Reply:
ok